pageserver/src/config.rs

```diff
@@ -731,6 +731,13 @@ impl PageServerConf {
                 })?);
         }
 
+        if let Some(eviction_policy) = item.get("eviction_policy") {
+            t_conf.eviction_policy = Some(
+                toml_edit::de::from_item(eviction_policy.clone())
```

Comment:
The other config keys are deserialized using hand-rolled FromStr's. Don't understand why.
Is this acceptable?
For the alternative hand-rolled variant, see the left-hand-side of https://github.com/neondatabase/neon/pull/3552/commits/15fb8a3824150d9c7c6b0c833f89603a52d2cb0b

Reply:
Note here we are deserializing it as toml, though I cannot find docs for `toml_edit::de::from_item`, nor the fn from their repository. Aah ok: https://docs.rs/toml_edit/0.17.1/toml_edit/de/fn.from_item.html (we are using a bit older version).

I think it makes sense to rely on serde here. I understand in the past this hasn't been done to give more clearer errors, because by default serde errors have not been helpful.

How are the errors looking for this one?

Reply:
Looked fine. I tried with a wrong `kind`, and it even spit out the list of valid `kind`s.

Reply:
Yes for that it should work nicely ... Can't remember for which it didn't work. Resolving!